background.js

```diff
@@ -1,33 +1,75 @@
-const showMessage = async (message) => {
+const setLoadingState = async (active) => {
+  const activeCss = `.caal-loading {
+    position: absolute;
+    display: flex;
+    background: #85a94e;
+    padding: 1em;
+
+    transform: translateX(-50%);
+    left: 50%;
+    margin-top: 40vh;
+
+    z-index: 2000;
+
+    box-shadow: 0 var(--borderHeight) 0 0 var(--borderColor),0 .7rem .95rem .05rem var(--secondaryBorderColor);
+    border-radius: var(--borderRadius,.5rem);
+
+    color: #fff;
+    font-family: var(--globalSecondaryFont);
+    font-size: 3rem;
+    line-height: 1;
+    font-weight: 700;
+    text-shadow: 0 .1rem 0 rgba(0,0,0,.4);
+  }`;
+
+  const deactiveCss = `.caal-loading {
+    display: none;
+  }`;
+
+  browser.tabs.insertCSS({ code: active ? activeCss : deactiveCss });
+};
+
+const fetchField = async (url, field) => {
+  const response = await fetch(url);
+  const data = await response.json();
+  return data[field];
+};
+
+const getPGN = async (playerName, gameUrl) => {
+  const archives = await fetchField(
+    `https://api.chess.com/pub/player/${playerName}/games/archives`,
+    "archives"
+  );
+
+  for (let i = archives.length - 1; i >= 0; i--) {
```

Comment:
What is the reason you're iterating from the last one to the first one?

There may be a good reason for it. It is simply not documented in the code, which could be confusing for other readers

Reply:
Archives are given from the oldest to the most recent games. Thus iterating from the last one (the most recent) is usually better, because players usually want to analyse their most recent games. So you will make less fetches. Comment about that was added

Reply:
Makes sense, great comment :ok_hand: